### PR TITLE
Expose database models on DataSources

### PR DESCRIPTION
**Pull request by willsb**
This will make consuming the database specifics possible (and cast free!).

Squash whenever

### CODE REVIEW

File: Toggl.Foundation/DataSources/TimeEntriesDataSource.cs [hunk 1]
```diff
@@ -45,18 +44,20 @@ public IObservable<Unit> Delete(long id)
...
-                .SetBillable(billable)
-                .SetIsDirty(true)
-                .Build()
-                .Apply(repository.Create)
-                .Do(safeSetCurrentlyRunningTimeEntry);
-                
-        public IObservable<ITimeEntry> Stop(DateTimeOffset stopTime)
+        public IObservable<IDatabaseTimeEntry> Start(DateTimeOffset startTime, string description, bool billable)
+        {
+            return idProvider.GetNextIdentifier()
```
Comment (simonrozsival) [inline Toggl.Foundation/DataSources/TimeEntriesDataSource.cs]: Why changing this from expression-bodied function to normal function?

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/MainViewModelTests.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿﻿﻿using System;
+﻿﻿﻿﻿using System;
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/MainViewModelTests.cs]: BOM?

File: Toggl.Foundation.Tests/DataSources/TimeEntriesDataSourceTests.cs [hunk 3]
```diff
@@ -6,16 +6,14 @@
 using FluentAssertions;
 using Microsoft.Reactive.Testing;
 using NSubstitute;
-using Toggl.Foundation.Models;
 using Toggl.Foundation.DataSources;
 using Toggl.Multivac.Models;
 using Toggl.PrimeRadiant;
 using Toggl.PrimeRadiant.Exceptions;
 using Toggl.PrimeRadiant.Models;
+using Toggl.Foundation.Models;
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/DataSources/TimeEntriesDataSourceTests.cs]: I liked it better when it was in lexical order :)

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/MainViewModelTests.cs (hunk 2 above)
Comment (willsb) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/MainViewModelTests.cs]: Probs

**Pull request merged**